Marlin/src/HAL/STM32/SoftwareSerial.cpp

```diff
@@ -137,12 +142,15 @@ void SoftwareSerial::setSpeed(uint32_t speed) {
           pre *= 2;
         }
       } while (cmp_value >= UINT16_MAX);
+      // Arduino_Core_STM32 1.8 requires setMode to be called on a channel to start the timer.
+      // This is fixed in the following PR, which has not yet been released.
+      // https://github.com/stm32duino/Arduino_Core_STM32/pull/849
+      timer.setMode(1, TIMER_OUTPUT_COMPARE, NC);
       timer.setPrescaleFactor(pre);
       timer.setOverflow(cmp_value);
       timer.setCount(0);
       timer.attachInterrupt(&handleInterrupt);
       timer.resume();
-      NVIC_SetPriority(timer_interrupt_number, timer_interrupt_priority);
```

Comment:
The required framework required us to reset the priority every time we resumed the timer, but that is no longer necessary. Setting it once in `setInterruptPriority` is adequate.